util/codec/decimal_test.go

```diff
@@ -48,7 +48,7 @@ func (s *testDecimalSuite) TestDecimalCodec(c *C) {
 		v := types.NewDecFromFloatForTest(input.Input)
 		datum := types.NewDatum(v)
 		b := EncodeDecimal([]byte{}, datum.GetMysqlDecimal(), datum.Length(), datum.Frac())
-		_, d, err := DecodeDecimal(b)
```

Comment:
add test for precision and frac here.

Reply:
If `DecodeDecimal` always retuned legitimate `precision` or `frace`, then there is no need to add such tests. Alternative way to test could be check the error which is retuned by `DecodeDecimal`.